### ISSUE TITLE
problem in vscode editor

### ISSUE THREAD

**Issue by udhayprakash** [Microsoft/vscode#16511]
- VSCode Version: 1.7.2
- OS Version: windows 7 sp2

Steps to Reproduce:

1. When I edited a _pre-saved file_, and **run the code**, it is resulting from the pre-saved version. It is not autosaving the edited file and running. 

__EDIT:__ Moving issue 2 as a separate issue.

**Comment by bpasero** [comment]
@udhayprakash can you please report issue 2 as separate issue?

As for 1: do I understand it right that you make changes to a file and then hit F5 to run it and the changes are not being picked up? What file type is this?

**Comment by isidorn** [comment]
2 is as designed, we have this limitation before but decided to lift it for various reasons

**Comment by udhayprakash** [comment]
@bpasero In general, _any IDE/editor_, when there are unsaved changes, and we run the script, it will autosave and then run the script/program. 

In vscode, I observed that if I don't save manually, the new changes are not autosaved. Rather, the last saved content is executed.  

In layman's words, **there isn't autosave and run, when we are executing the script.** 

I noticed it when working with **python scripts**. 


And, I am moving issue 2, as a separate issue as per your suggestion @bpasero . @isidorn

**Comment by bpasero** [comment]
@udhayprakash but the file is being saved when you run, isn't it? the dirty decoration goes away right?

**Comment by udhayprakash** [comment]
@bpasero unfornutely the file is not getting saved automatically, when we are executing it.
![image](https://cloud.githubusercontent.com/assets/11205605/20925115/de118200-bbda-11e6-993b-84092d88313e.png)

**Comment by bpasero** [comment]
@udhayprakash how do you run your application?

**Comment by udhayprakash** [comment]
@bpasero I used either of the highlighted options, as illustrated in the screenshot
![image](https://cloud.githubusercontent.com/assets/11205605/21173741/35eb916c-c1ff-11e6-8776-5abf29f4c3c2.png)

**Comment by bpasero** [comment]
Reading your description it seems that the issue originates from an extension and not VS Code itself. Thereby I will close this issue and kindly ask you to report this issue against the extension repository. 

Most extensions are on GitHub as well so it should be easy to file it against the extension. If you open an extension on the market place, click on the **Support** link to find out more.

<img width="196" alt="screen shot 2016-05-05 at 12 51 17" src="https://cloud.githubusercontent.com/assets/900690/15041943/263e2282-12c0-11e6-9a33-d2b3b500d61b.png">

Thanks again for your feedback!

**Comment by udhayprakash** [comment]
Then, why "Run Python File in Terminal" is also not working. It isn't related to addon @bpasero

**Comment by bpasero** [comment]
@udhayprakash I am quite sure that we are not providing a "Run Python File in Terminal" out of the box (/cc @rebornix )

**Comment by rebornix** [comment]
@udhayprakash you might be using https://marketplace.visualstudio.com/items?itemName=formulahendry.code-runner
